pkg/operator/controller/ingress/controller_test.go

```diff
@@ -4,10 +4,421 @@ import (
+					Type: operatorv1.LoadBalancerServiceStrategyType,
+					LoadBalancer: &operatorv1.LoadBalancerStrategy{
+						Scope: operatorv1.ExternalLoadBalancer,
+					},
+				},
+			},
+		}
+		ingressControllerWithHostNetwork = &operatorv1.IngressController{
+			Status: operatorv1.IngressControllerStatus{
+				EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
+					Type: operatorv1.HostNetworkStrategyType,
+					HostNetwork: &operatorv1.HostNetworkStrategy{
+						Protocol: operatorv1.TCPProtocol,
+					},
+				},
+			},
+		}
+		makeInfra = func(platform configv1.PlatformType) *configv1.Infrastructure {
+			return &configv1.Infrastructure{
+				Status: configv1.InfrastructureStatus{
+					Platform: platform,
+				},
+			}
+		}
+	)
+
+	testCases := []struct {
+		name        string
+		infraConfig *configv1.Infrastructure
+		expectedIC  *operatorv1.IngressController
+	}{
+		{
+			name:        "Alibaba",
+			infraConfig: makeInfra(configv1.AlibabaCloudPlatformType),
+			expectedIC:  ingressControllerWithLoadBalancer,
+		},
+		{
+			name:        "AWS",
+			infraConfig: makeInfra(configv1.AWSPlatformType),
+			expectedIC:  ingressControllerWithLoadBalancer,
+		},
+		{
+			name:        "Azure",
+			infraConfig: makeInfra(configv1.AzurePlatformType),
+			expectedIC:  ingressControllerWithLoadBalancer,
+		},
+		{
+			name:        "Bare metal",
+			infraConfig: makeInfra(configv1.BareMetalPlatformType),
+			expectedIC:  ingressControllerWithHostNetwork,
+		},
+		{
+			name:        "Equinix Metal",
+			infraConfig: makeInfra(configv1.EquinixMetalPlatformType),
+			expectedIC:  ingressControllerWithHostNetwork,
+		},
+		{
+			name:        "GCP",
+			infraConfig: makeInfra(configv1.GCPPlatformType),
+			expectedIC:  ingressControllerWithLoadBalancer,
+		},
+		{
+			name:        "IBM Cloud",
+			infraConfig: makeInfra(configv1.IBMCloudPlatformType),
+			expectedIC:  ingressControllerWithLoadBalancer,
+		},
+		{
+			name:        "No platform",
+			infraConfig: makeInfra(configv1.NonePlatformType),
+			expectedIC:  ingressControllerWithHostNetwork,
+		},
+		{
+			name:        "OpenStack",
+			infraConfig: makeInfra(configv1.OpenStackPlatformType),
+			expectedIC:  ingressControllerWithHostNetwork,
+		},
+		{
+			name:        "Power VS",
+			infraConfig: makeInfra(configv1.PowerVSPlatformType),
+			expectedIC:  ingressControllerWithLoadBalancer,
+		},
+		{
+			name:        "RHV",
+			infraConfig: makeInfra(configv1.OvirtPlatformType),
+			expectedIC:  ingressControllerWithHostNetwork,
+		},
+		{
+			name:        "vSphere",
+			infraConfig: makeInfra(configv1.VSpherePlatformType),
+			expectedIC:  ingressControllerWithHostNetwork,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ic := &operatorv1.IngressController{}
+			infraConfig := tc.infraConfig.DeepCopy()
+			if actualResult := setDefaultPublishingStrategy(ic, infraConfig); actualResult != true {
```

Comment:
Using coverage I don't see a case for: 

```go
                case configv1.LibvirtPlatformType:
                        strategyType = operatorv1.HostNetworkStrategyType
```

in `setDefaultPublishingStrategy()`.

Reply:
I could add a case for Libvirt, but it isn't a supported platform, so I didn't see a need.

Reply:
We do specify the defaulting behavior for Libvirt in the IngressController API godoc though, so I'll add a case.